<!-- Phidgets Magnetometer launch file -->

<launch>

  #### Nodelet manager ######################################################

  <node pkg="nodelet" type="nodelet" name="mag_manager"
    args="manager" output="screen" />

  #### Magnetometer Driver ##################################################

  <node pkg="nodelet" type="nodelet" name="PhidgetsMagnetometerNodelet"
    args="load phidgets_magnetometer/PhidgetsMagnetometerNodelet mag_manager"
    output="screen">

    # See README.md for more information on the parameters

    # optional param serial, default is -1
    <!-- <param name="serial" value="123456"/> -->

    # optional param hub_port, used if connected to a VINT hub
    <!-- <param name="hub_port" value="0"/> -->

    # optional param frame_id, default is "imu_link"
    <!-- <param name="frame_id" value="imu_link"/> -->

    # optional param magnetic_field_stdev, default is 1.1 milligauss
    <!-- <param name="magnetic_field_stdev" type="double" value="1.1e-7" /> -->

    # supported data rates: 4 8 16 24 32 40 ... 1000 (in ms)
    <param name="data_interval_ms" value="4"/>

    # optional param publish_rate, defaults to 0
    <!-- <param name="publish_rate" value="0" /> -->

    # optional param server_name, required to locate remote network servers
    <!-- <param name="server_name" value="myPhidgetServer" /> -->

    # optional param server_ip, required to locate remote network servers
    <!-- <param name="server_ip" value="0.0.0.0" /> -->

    # compass correction params (see http://www.phidgets.com/docs/1044_User_Guide)
    <!-- <param name="cc_mag_field" value="0.52859"/>
         <param name="cc_offset0" value="0.03921"/>
         <param name="cc_offset1" value="0.19441"/>
         <param name="cc_offset2" value="-0.03493"/>
         <param name="cc_gain0" value="1.81704"/>
         <param name="cc_gain1" value="1.81028"/>
         <param name="cc_gain2" value="2.04819"/>
         <param name="cc_t0" value="0.00142"/>
         <param name="cc_t1" value="-0.03591"/>
         <param name="cc_t2" value="0.00160 "/>
         <param name="cc_t3" value="-0.05038"/>
         <param name="cc_t4" value="-0.03942"/>
         <param name="cc_t5" value="-0.05673"/> -->
  </node>

</launch>
